### PR TITLE
Add Makie package extension

### PR DESCRIPTION
**Pull request by andrewwinters5000**
Create a [package extension](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) for the plotting functionality implemented herein that requires Makie.jl to be loaded. To remain backwards compatible with Julia v1.6, I've kept the current approach that is based on Requires.jl.

When a Julia version above 1.9.1 is used, the package extension is loaded instead of relying on Requires.jl. The only user-visible difference is that we define "empty" functions for `plotProject!` and `updatePlot!` in HOHQMesh.jl that are then exported, since the package extension cannot export anything itself.

I tested it locally using Julia versions 1.10.0 and 1.8.5 and everything passed. ~I was not sure where the docstrings for the functions should live. Right now they are on the actual functions contained in the package extension but they might need moved.~ I had to place the docstings above the functions in `src/Viz/visualization.jl` for them to show up properly in the docs. So, I added a note in the docstrings pointing to where the full function is actually implemented.

### CODE REVIEW

File: src/Viz/visualization.jl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+# Convenience constants to make plotting easier
+const MODEL = 1; const GRID = 2; const MESH = 4; const EMPTY = 0
+const REFINEMENTS = 8; const ALL = 15
```
Comment (ranocha) [inline src/Viz/visualization.jl:4]: Where were these defined before?

File: src/HOHQMesh.jl [hunk 2]
```diff
@@ -7,25 +7,19 @@ using HOHQMesh_jll: HOHQMesh_jll
 using Requires: @require
 
 function __init__()
-  # Enable features that depend on the availability of the Makie package
-  @require Makie="ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a" begin
-    using .Makie
-    if isdefined(Makie, :to_textsize)
-      @warn "You seem to be using an older version of Makie (< v0.19). Some plotting functions may not work."
```
Comment (ranocha) [inline src/HOHQMesh.jl:14]: The Makie compat setting requires v0.20 instead of v0.19 mentioned here. Is that correct?

File: src/Viz/visualization.jl (hunk 1 above)
Comment (andrewwinters5000) [reply src/Viz/visualization.jl:4]: These used to be at the top of the file `VizProject.jl` but I had to move them because the package extension cannot export these constants.

File: src/HOHQMesh.jl (hunk 2 above)
Comment (andrewwinters5000) [reply src/HOHQMesh.jl:14]: The Makie v0.20 is the one that deprecated a keyword argument that was removed in #50 . I can bump this back down to v0.19 to see what happens, but I expect some warning will get thrown.

Comment (andrewwinters5000) [reply src/HOHQMesh.jl:14]: As expected, if I set the compat bound to v0.19 (or lower) the precompilation as well as when I run and visualize the interactive tool throw some deprecation warnings. Everything still works however.

File: src/Viz/visualization.jl (hunk 1 above)
Comment (ranocha) [reply src/Viz/visualization.jl:4]: I asked because GitHu just told me that `VizProject.jl` was moved to another folder without modifying its content. What you wrote is fine 👍

File: src/Viz/visualization.jl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Example: To plot the model and the grid, `plotOptions = MODEL + GRID`. To plot
+just the mesh, `plotOptions = MESH`.
+
+To plot everything, `plotOptions = MODEL + GRID + MESH + REFINEMENTS`
+
+Contents are overlaid in the order: GRID, MESH, MODEL, REFINEMENTS
+!!! note
+    The function implementation is found in `ext/VizProject.jl`.
+"""
+function plotProject! end
```
Comment (ranocha) [inline src/Viz/visualization.jl]:
```suggestion
Contents are overlaid in the order: `GRID`, `MESH`, `MODEL`, `REFINEMENTS`
"""
function plotProject! end
# Note: The function implementation is found in `ext/VizProject.jl`.
```

File: src/Viz/visualization.jl [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Replot with the new plotOptions = combinations (sums) of
+
+    GRID, MESH, MODEL, REFINEMENTS
+
+Example: updatePlot!(p, MESH + MODEL)
+
+!!! note
+    The function implementation is found in `ext/VizProject.jl`.
+"""
+function updatePlot! end
```
Comment (ranocha) [inline src/Viz/visualization.jl]:
```suggestion
Example: `updatePlot!(p, MESH + MODEL)`
"""
function updatePlot! end
# Note: The function implementation is found in `ext/VizProject.jl`.
```

File: test/test_visualization.jl [hunk 5]
```diff
@@ -87,8 +87,6 @@ using CairoMakie
     meshFileFormat = getMeshFileFormat(p_visu)
     setFileNames!(p_visu, meshFileFormat)
 
-    @test_nowarn updatePlot!(p_visu)
-
```
Comment (ranocha) [inline test/test_visualization.jl]: Why did you remove this test?

File: src/Viz/visualization.jl (hunk 3 above)
Comment (ranocha) [reply src/Viz/visualization.jl]: The note was just an internal comment, not something that should be shown to users

File: src/HOHQMesh.jl (hunk 2 above)
Comment (ranocha) [reply src/HOHQMesh.jl:14]: You could change the compat setting in Project.toml to `0.19, 0.20` - but I'm fine keeping it as it is right now in the PR

Comment (andrewwinters5000) [reply src/HOHQMesh.jl:14]: Yeah, that was my mistake. I forgot to do some clean-up in my initial commits.

File: test/test_visualization.jl (hunk 5 above)
Comment (andrewwinters5000) [reply test/test_visualization.jl]: The function `updatePlot!` with only one input argument was legacy code. Internally it was never used for anything. Although now that I write this, it makes this PR a breaking change if I completely remove it. I will put it back in to avoid this.

File: src/Viz/visualization.jl [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Plot objects specified by the `plotOptions`. Construct the `plotOptions` by the sum
+of what is to be drawn from the choices `MODEL`, `GRID`, `MESH`, `REFINEMENTS`.
+
+Example: To plot the model and the grid, `plotOptions = MODEL + GRID`. To plot
+just the mesh, `plotOptions = MESH`.
+
+To plot everything, `plotOptions = MODEL + GRID + MESH + REFINEMENTS`
+
+Contents are overlaid in the order: `GRID`, `MESH`, `MODEL`, `REFINEMENTS`
```
Comment (sloede) [inline src/Viz/visualization.jl:21]:
```suggestion
Contents are overlaid in the order: `GRID`, `MESH`, `MODEL`, `REFINEMENTS`

!!! note "Requires Makie.jl"
    Please note that for this function to work, you need to load Makie.jl
    in your REPL (e.g., by calling `using GLMakie`).
```
Not sure if this is all correct, but I would inform the user about the Makie requirement here in the docstring.

File: src/Viz/visualization.jl [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+
+"""
+    updatePlot!(proj::Project, plotOptions::Int)
+
+Replot with the new plotOptions = combinations (sums) of
+
+    GRID, MESH, MODEL, REFINEMENTS
+
+Example: `updatePlot!(p, MESH + MODEL)`
```
Comment (sloede) [inline src/Viz/visualization.jl:38]:
```suggestion
Example: `updatePlot!(p, MESH + MODEL)`

!!! note "Requires Makie.jl"
    Please note that for this function to work, you need to load Makie.jl
    in your REPL (e.g., by calling `using GLMakie`).
```
Not sure if this is all correct, but I would inform the user about the Makie requirement here in the docstring.

**Review by sloede — CHANGES_REQUESTED** [review]
A small suggestion, but otherwise LGTM

File: src/Viz/visualization.jl (hunk 6 above)
Comment (andrewwinters5000) [reply src/Viz/visualization.jl:21]: I have never tried it with `using Makie`. I would probably indicate that one needs a Makie backend. So, I would change the last line to be 
```
(e.g., by calling `using GLMakie`).
```

Comment (sloede) [reply src/Viz/visualization.jl:21]: Yes, totally! I've updated my suggestion accordingly

**Review by ranocha — APPROVED** [review]

**Review by sloede — APPROVED** [review]
